### PR TITLE
ENH: natively support Graph in getisord and gamma modules

### PR DESCRIPTION
**Pull request by martinfleis**
I have vectorized Gamma computation which is now about 10x faster on a dummy lat2W(40, 40) graph.

### CODE REVIEW

File: esda/gamma.py
Comment (martinfleis) [inline esda/gamma.py:267]: What is the expectation for a custom function here? It is undocumented. I see how it is defined in tests but is that the expectation? Is some downstream code relying on this?

Comment (sjsrey) [reply esda/gamma.py:267]: I don't think any downstream code relies on this.

This is handling the `operation` option that specifies the form of the attribute similarity measure, which is documented in the class docstring. But maybe you are referring to something else not being documented?

Comment (martinfleis) [reply esda/gamma.py:267]: I see the example but I wasn't sure what the expectations for the function signature are. I can derive that from the example but I'm a bit hesitant trying to replicate that as the function optimised for graph would probably look different.

Comment (sjsrey) [reply esda/gamma.py:267]: I don't think there were any expectations. More like a place holder in case other similarity functions were proposed.

On a related issue, would it be possible to add some helper methods on the graph to simplify these: https://github.com/pysal/esda/blob/9987fd4f63b2857fcfe465e22436913d0f54b10c/esda/gamma.py#L248-L249

Would that make sense?

Comment (martinfleis) [reply esda/gamma.py:267]: Yeah, I've been writing that way too often recently so it certainly deserves a helper method.

**Review by knaaptime — APPROVED** [review]

**Comment by martinfleis** [comment]
I got no clue why that one test in that one specific env is failing...

**Comment by martinfleis** [comment]
I can't reproduce that issue on neither macOS or Linux. @jGaboardi any suggestions?

**Comment by jGaboardi** [comment]
Maybe the warning was coming from an older version of something and it's no longer being thrown since bumped up oldest?

**Comment by martinfleis** [comment]
It is being emitted in every other env apart from this one.

**Comment by jGaboardi** [comment]
Also, can't find anything at all that would lead to that logic path. Very strange...

**Comment by martinfleis** [comment]
Magic... all green :D

**Review by jGaboardi — APPROVED** [review]

**Comment by jGaboardi** [comment]
* CI is all green - `libpysal` is pinned - we have approvals
* I'll merge tomorrow in the AM unless someone else already has or there are detractors

**Pull request merged**
